File: fastapi/app/schemas/ingredientRate.py
from sqlalchemy import Column, Integer, Float, Date, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.ext.declarative import declarative_base
from pydantic import BaseModel
from typing import Optional
from .base import Base

# SQLAlchemy ORM 모델
class IngredientRate(Base):
    __tablename__ = 'ingredientrate'

    ingredient_rate_id = Column(Integer, primary_key=True, autoincrement=True, nullable=False)
    date = Column(Date, nullable=False)
    week_increase_rate = Column(Float, nullable=False, name="week_increase_rate")
    month_increase_rate = Column(Float, nullable=False, name="month_increase_rate")
    year_increase_rate = Column(Float, nullable=False, name="year_increase_rate")
    week_increase_price = Column(Integer, nullable=False, name="week_increase_price")
    month_increase_price = Column(Integer, nullable=False, name="month_increase_price")
    year_increase_price = Column(Integer, nullable=False, name="year_increase_price")
    
    # Ingredient와의 관계 설정
    ingredient_id = Column(Integer, ForeignKey('ingredient.ingredient_id'), nullable=False)  # Foreign key
    ingredient = relationship("Ingredient", back_populates="ingredientrate")  # Relationship 설정

    def __init__(self, date: str, week_increase_rate: float, 
                 week_increase_price: int, ingredient_id: int, 
                 month_increase_rate: float, month_increase_price: int,
                 year_increase_rate: float, year_increase_price: int):
        self.date = date
        self.week_increase_rate = week_increase_rate
        self.week_increase_price = week_increase_price
        self.ingredient_id = ingredient_id
        self.month_increase_price = month_increase_price
        self.month_increase_rate=month_increase_rate
        self.year_increase_price=year_increase_price
        self.year_increase_rate=year_increase_rate

# Pydantic 스키마
class IngredientRateSchema(BaseModel):
    ingredient_rate_id: Optional[int]
    date: str  # LocalDate를 문자열로 처리
    week_increase_rate: float
    week_increase_price: int
    month_increase_rate: float
    month_increase_price: int
    year_increase_rate: float
    year_increase_price: int
    ingredient_id: int

    class Config:
        from_attributes = True

File: backend/mijung/src/main/java/com/example/mijung/ingredient/entity/IngredientRate.java
package com.example.mijung.ingredient.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.time.LocalDate;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class IngredientRate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ingredient_rate_id", updatable = false, nullable = false)
    private Integer id;

    @Column(nullable = false)
    private LocalDate date;

    @Column(name = "week_increase_rate", nullable = false)
    private Float weekIncreaseRate;

    @Column(name = "month_increase_rate", nullable = false)
    private Float monthIncreaseRate;

    @Column(name = "year_increase_rate", nullable = false)
    private Float yearIncreaseRate;

    @Column(name = "week_increase_price", nullable = false)
    private Integer weekIncreasePrice;

    @Column(name = "month_increase_price", nullable = false)
    private Integer monthIncreasePrice;

    @Column(name = "year_increase_price", nullable = false)
    private Integer yearIncreasePrice;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ingredient_id", nullable = false)
    private Ingredient ingredient;

    @Builder
    public IngredientRate(Integer id, LocalDate date, Float weekIncreaseRate, Integer weekIncreasePrice,
                          Ingredient ingredient) {
        this.id = id;
        this.date = date;
        this.weekIncreaseRate = weekIncreaseRate;
        this.weekIncreasePrice = weekIncreasePrice;
        this.ingredient = ingredient;
    }

    ;
}
